utils/array.js

```diff
@@ -2,7 +2,7 @@ import { xor } from 'lodash';
 import { get } from '@/utils/object';
 
 export function removeObject(ary, obj) {
-  const idx = ary.indexOf(obj);
+  const idx = ary.findIndex(resource => resource.id === obj.id);
```

Comment:
This is for #604 , specifically deletion not being reflected in the resource table. Seems like the remove mutation is sometimes calling removeObject with an object that doesn't reference something in the store but is equal to it

Reply:
There is supposed to be exactly one model object in the store for a particular type/id.  If that's not true in some case we need to find and address the root of that problem; finding a different object that is "similar" is just masking the problem and will pop up in other places (like edit/change events instead of remove).